Geode.Angular/src/app/app.module.ts

```diff
@@ -2,29 +2,58 @@ import { NgModule } from '@angular/core';
 import { AgGridAngular } from 'ag-grid-angular'
 
 import { RouterOutlet } from '@angular/router';
-import { HttpClientModule} from '@angular/common/http';
+import { HTTP_INTERCEPTORS, HttpClientModule} from '@angular/common/http';
 import { BrowserModule } from '@angular/platform-browser';
 import { AppRoutingModule } from './app-routing.module';
 
 import { AppComponent } from './app.component';
-import { NavbarComponent } from './shared/navbar/navbar.component';
-import { UsersComponent } from './users/users.component';
-import { AuthModule } from './auth/auth.module';
+import { NavbarComponent } from './layout/navbar/navbar.component';
+import { UsersComponent } from './pages/users/users.component';
+import { AuthInterceptor } from './interceptors/auth.interceptor';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatExpansionPanel, MatExpansionPanelHeader, MatExpansionPanelTitle } from '@angular/material/expansion';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatFormField, MatLabel } from '@angular/material/form-field';
+import { MatOption, MatSelect } from '@angular/material/select';
+import { MatCheckbox } from '@angular/material/checkbox';
+import { MatInput } from '@angular/material/input';
+import { FilterComponent } from './shared/components/filter/filter.component';
+import { AuthModule } from './pages/auth/auth.module';
 
 @NgModule({
   declarations: [
     AppComponent,
     NavbarComponent,
     UsersComponent,
+    FilterComponent
   ],
   imports: [
     RouterOutlet,
     HttpClientModule,
     BrowserModule,
+    BrowserAnimationsModule,
     AppRoutingModule,
     AgGridAngular,
     AuthModule,
+    FormsModule,
```

Comment:
All these imports should be in `shared.module.ts` , also you should export them to use. 
After that you have to import only SharedModule inside the component you need